packages/nc-gui/components/virtual-cell/Rollup.vue

```diff
@@ -2,10 +2,54 @@
 import { CellValueInj, inject } from '#imports'
 
 const value = inject(CellValueInj)
+
+const showEditWarning = ref(false)
+
+const showClearWarning = ref(false)
+
+const showEditWarningMessage = () => {
+  showEditWarning.value = true
+
+  setTimeout(() => {
+    showEditWarning.value = false
+  }, 3000)
+}
+
+const showClearWarningMessage = () => {
+  showClearWarning.value = true
+
+  setTimeout(() => {
+    showClearWarning.value = false
+  }, 3000)
+}
```

Comment:
Use [`refAutoReset`](https://vueuse.org/shared/refAutoReset/) instead 😉 

```ts
const timeout = 3000 // in ms

const showEditWarning = refAutoReset(false, timeout)
const showClearWarning = refAutoReset(false, timeout)

useSelectedCellKeyupListener(inject(ActiveCellInj, ref(false)), (e: KeyboardEvent) => {
  switch (e.key) {
    case 'Enter':
      showEditWarning.value = true // will reset after 3000ms
      break
    case 'Delete':
      showClearWarning.value = true // will reset after 3000ms
      break
  }
})

Reply:
That's a nice utility composable :+1: